frontend/src/components/elements/DataFrame/DataFrame.tsx

```diff
@@ -285,7 +280,7 @@ class DataFrame extends React.PureComponent<Props, State> {
       rowHeight,
       headerHeight,
       border,
-      height,
+      height: Math.min(rows * rowHeight + border, height || 300),
```

Comment:
So a dataframe doesn't expand to full screen height unless its height is taller than fullscreen?

Reply:
I don't know if you need `|| 300` since you set a default value for height in defaultProps

Reply:
`st.dataframe()` takes a `height` but I don't see it being used at all here

Reply:
defaultProps is not working for TS, so I created DEFAULT_HEIGHT now. And that line is fine, because for DataFrames we don't need expand if the table does not need it.